### ISSUE TITLE
feat(ball_detection): DINOv3 patch tokenと3軸RoPEによる時系列検出モデルを実装する

### ISSUE THREAD

**Issue by Motoki0705** [Motoki0705/tennis-lab#551]
## 概要

DINOv3 ViT-B/16のpatch tokenを全時刻・全空間で処理し、`(time, y, x)`の3軸RoPEを適用するbidirectional Transformer型ball detectionモデルを実装する。

## 課題

- 現行の`conv_next_unet`はMDD入力に依存し、`T=1`の静止画学習へ直接適用できない。
- `dino_pseudo3d`は旧DINO backboneとConv3d decoderの構成で、DINOv3 patch token、global self-attention、3軸RoPEを利用していない。
- Phase 1の`T=1` checkpointを、モデル重みの再構成なしでPhase 2の可変`T`へ移行できるモデル契約が必要。

## 提案

以下のモデルを`src/tasks/ball_detection/models/`へ追加する。

```text
(B, T, 3, 288, 512)
  -> frame-wise DINOv3 ViT-B/16
  -> patch tokens (B, T, 18, 32, C_backbone)
  -> decoder projection
  -> flatten (B, T*18*32, D)
  -> bidirectional TransformerBlock x L
       Q/Kへ(time, y, x)の3軸RoPE
  -> restore (B, T, D, 18, 32)
  -> frame-shared heatmap head
  -> logits (B, 1, T, 288, 512)
```

- decoderは全patch tokenをself-attentionし、learned object queryやcross-attentionは使用しない。
- `src/utils/models/components/rope.py`へローカル整数座標`(t, y, x)`を渡す。FPSやtimestampは入力しない。
- attentionは非causalのbidirectional方式とする。
- heatmap headは各時刻に同一重みを適用し、`18x32`から`288x512`へ復元する。
- DINOv3 loaderを`src/utils/models/loading/`へ共有化し、court detectionと同じロード・strict検証を利用する。
- decoder構成、RoPE、dropout、backbone学習範囲をHydra configから設定可能にする。

## スコープ

対象:

- DINOv3 ViT-B/16 patch-token encoder adapter
- 3軸RoPE付きTransformer decoder
- `288x512` heatmap head
- `T=1`と可変`T`のcheckpoint互換
- `frozen`、`last_n_blocks`、`full`を表現できるbackbone設定
- shape、RoPE座標、gradient、checkpointロードのunit test

対象外:

- dataset変換、window生成、static/temporal sampling
- 学習戦略の選定と実験
- 複数checkpointの自動評価・比較レポート
- causal/online inference、FPS・timestamp位置埋め込み、DETR型query

## 受け入れ条件

- `model.num_frames=1`と複数frame設定で、入力`(B,T,3,288,512)`からlogits`(B,1,T,288,512)`を生成できる。
- 全decoder layerのQ/Kへ`(time, y, x)`の3軸RoPEが適用され、token順序と座標対応がテストされている。
- Phase 1の`T=1` checkpointを、同一モデルパラメータのままPhase 2の`T>1`設定へstrictにロードできる。
- DINOv3 loaderが共有moduleから利用され、backbone checkpointの不整合を明示的に検出できる。
- model configだけでdecoder規模、RoPE、backbone学習範囲を変更できる。
- forward/backwardのsmoke testと主要shapeのunit testが通る。

## 補足

- 親Issue: #549
- 入力データ契約は別の子Issueで整備する。
- `T=8`ではtoken数が`4,608`になるため、VRAM不足時はdecoder規模やgradient checkpointingをconfigで調整できる設計にする。
